### ISSUE TITLE
Are banners paused when hidden?

### ISSUE THREAD

**Issue by ecv80** [googleads/googleads-mobile-unity#1006]
Hi,

It seems the AdMob API provides a way to pause and resume ads. 

In my current implementation I'm hiding all banners on scene switch and showing whatever banner at scene start as a way to save time in instantiating a new banner view and loading ads. Then I load ads every 5 minutes for banners should the AdMob dashboard banner refresh rate not be enough for this purpose.

But my question is whether hidden banners are still running behind the scenes and therefore affecting performance during game play while they're hidden. If so, can we have a hook to pause and resume in AdMob API?

Also, could you check on my previous issue https://github.com/googleads/googleads-mobile-unity/issues/805 ?

Thanks!

**Comment by UragayalaDeepika** [comment]
@ecv80 Currently, the Unity Plugin does not provide an API to pause or resume the Banner AdView. We do have these APIs in the Android native SDK and something the Unity plugin currently exposes. When the BannerView is hidden, the impression ping is not fired to the user but the received creative is still held in the memory. So my advice would be to monitor the number of prefetched ads being held in memory and manage accordingly. The creative will always be held in memory as long as the BannerView reference is valid.

**Comment by ecv80** [comment]
Hi @UragayalaDeepika , sorry for the delay and thanks for the comprehensive answer.
However I don't understand much of the jargon being utilized nor anything about the native API.

If I understand correctly, banners aren't being paused when hidden, regardless of whether the pausing and resuming functionalities are being exposed in the Unity plugin (which they aren't, as you said).

If so, in layman words: are hidden banners running behind the scenes in a way that will significantly impact game performance and drop frame rate?

**Comment by stowy** [comment]
@ecv80 - banners in both IOS and Android both use web views, and the underlying behavior of the banner in each environment is dictated by the webview implementation of that platform.

That said, when banners are hidden in Android, the webview is paused, see [this line of code](https://github.com/googleads/googleads-mobile-unity/blob/f799a2deb34c6f7921779107bb6ecef901af8c83/source/android-library/app/src/main/java/com/google/unity/ads/Banner.java#L333).

It is not expected that the banner view would affect game performance in terms of CPU usage when hidden, however having many banner views instantiated would add memory pressure which could have adverse impacts.

**Comment by ecv80** [comment]
Thanks for your excellent answer!
